sdk/cognitivelanguage/azure-ai-textanalytics/azure/ai/language/text/aio/_patch.py

```diff
@@ -0,0 +1,362 @@
+                next_link = s.next_link
+                actions_payload: TextActions = s.actions
+                return next_link, [actions_payload]
+
+            async def get_next(token: Optional[str]) -> Optional[AnalyzeTextOperationState]:
+                if token is None:
+                    return state
+                if not token:
+                    return None
+                return await _fetch_state_by_next_link(token)
+
+            return AsyncItemPaged(get_next, extract_data)
+
+        poller_holder: Dict[str, AnalyzeTextAsyncLROPoller[AsyncItemPaged["TextActions"]]] = {}
+
+        def get_long_running_output(pipeline_response):
+            final = pipeline_response.http_response
+            if final.status_code == 200:
+                data = json.loads(final.text())
+                op_state = AnalyzeTextOperationState(data)
+
+                poller_ref = poller_holder["poller"]
+                poller_ref._record_state_for_details(op_state) # pylint:disable=protected-access
+
+                paged = _build_pager_from_state(op_state)
+                return cls(pipeline_response, paged, {}) if cls else paged
+            raise HttpResponseError(response=final)
+
+        if polling is True:
+            polling_method: AsyncPollingMethod[AsyncItemPaged["TextActions"]] = cast(
+                AsyncPollingMethod[AsyncItemPaged["TextActions"]],
+                AsyncLROBasePolling(lro_delay, path_format_arguments=path_format_arguments, **kwargs),
+            )
+        elif polling is False:
+            polling_method = cast(AsyncPollingMethod[AsyncItemPaged["TextActions"]], AsyncNoPolling())
+        else:
+            polling_method = cast(AsyncPollingMethod[AsyncItemPaged["TextActions"]], polling)
+
+        if cont_token:
+            return AnalyzeTextAsyncLROPoller[AsyncItemPaged["TextActions"]].from_continuation_token(
+                polling_method=polling_method,
+                continuation_token=cont_token,
+            )
+
+        initial_kwargs = dict( # pylint:disable=use-dict-literal
+            text_input=text_input,
+            actions=actions,
+            display_name=display_name,
+            default_language=default_language,
+            cancel_after=cancel_after,
+            content_type=content_type,
+            cls=lambda x, y, z: x,  # passthrough raw pipeline response
+            headers=_headers,
+            params=_params,
+            **kwargs,
+        )
+        if body is not _Unset and body is not None:
+            initial_kwargs["body"] = body
+
+        raw_result = await self._analyze_text_job_initial(**initial_kwargs)
+        await raw_result.http_response.read()  # type: ignore[attr-defined]
+
+        lro: AnalyzeTextAsyncLROPoller[AsyncItemPaged["TextActions"]] = AnalyzeTextAsyncLROPoller(
+            self._client, raw_result, get_long_running_output, polling_method
+        )
+        poller_holder["poller"] = lro
```

Comment:
I think this code could be refactored - it should be relatively straightforward to patch a poller to expose the `details` property without such a complex patch. Maybe it's the `ItemPaged` that's making it complicated.... but can we open an issue to revisit this?